File: app/src/main/java/com/example/kiit/health/login/MainActivity.java
package com.example.kiit.health.login;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import com.example.kiit.health.R;

public class MainActivity extends AppCompatActivity {

    private final int SPLASH_DISPLAY_LENGTH = 4000;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.mainsplash);

        new Handler().postDelayed(new Runnable(){
            @Override
            public void run() {
                /* Create an Intent that will start the Menu-Activity. */
                Intent mainIntent = new Intent(MainActivity.this,mainac.class);
                MainActivity.this.startActivity(mainIntent);
                MainActivity.this.finish();
            }
        }, SPLASH_DISPLAY_LENGTH);
    }
}

File: Geek Hub/ProjectDirectory/SourceCode/app/src/main/java/com/android/qhackfst/ui/MainActivity.kt
package com.android.qhackfst.ui

import android.content.Intent
import android.os.Bundle
import android.provider.MediaStore
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.findNavController
import androidx.navigation.ui.setupWithNavController
import com.android.qhackfst.R
import com.android.qhackfst.ui.home.adult.DashboardViewModel
import com.android.qhackfst.ui.home.adult.UploadState
import com.android.qhackfst.util.tempFile
import com.google.android.material.bottomnavigation.BottomNavigationView

private const val RC_UPLOAD_FILE = 102

class MainActivity : AppCompatActivity() {


    private val dashboardViewModel by lazy {
        ViewModelProvider(this).get(DashboardViewModel::class.java)
    }

    var isStarted = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
        val navView: BottomNavigationView = findViewById(R.id.nav_view)

        val navController = findNavController(R.id.nav_host_fragment)


        navView.setupWithNavController(navController)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent?) {
        super.onActivityResult(requestCode, resultCode, intent)
        if (resultCode == RESULT_OK && requestCode==RC_UPLOAD_FILE) {
            val pair =
                tempFile(intent!!.data!!) ?: return
            dashboardViewModel.upload(pair)
        }
    }

    override fun onResume() {
        super.onResume()
        isStarted = false
        dashboardViewModel.uploadStatus.observe(this, Observer {
            when (it) {
                is UploadState.ChooseFile -> {
                    if (!isStarted) {
                        val intent =
                            Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI)
                        dashboardViewModel.uploadStatus.postValue(UploadState.FileNotChosen)
                        startActivityForResult(intent, RC_UPLOAD_FILE)
                        isStarted = true
                    }
                }
            }
        })

    }
}